packages/bytecode-utils/src/lib/bytecode.ts

```diff
@@ -32,92 +55,291 @@ export const decode = (bytecode: string): DecodedObject => {
   }
 
   // split auxdata
-  const [, auxdata] = splitAuxdata(bytecode);
+  const [, auxdata] = splitAuxdata(bytecode, auxdataStyle);
+  // See more here: https://github.com/vyperlang/vyper/pull/3010
+  if (auxdataStyle === AuxdataStyle.VYPER) {
```

Comment:
Also I wonder is it right that we only check for `AuxdataStyle.VYPER`? If we call this function with `VYPER_LT_0_3_10` or `VYPER_LT_0_3_5`, it would fail.

Reply:
`export const decode = <T extends AuxdataStyle.SOLIDITY | AuxdataStyle.VYPER>(`

`decode(` doesn't support all `AuxdataStyle` options

Reply:
Ah I missed that